src/controller/userController.js

```diff
@@ -47,6 +51,112 @@ class UserController {
       });
     }
   }
+
+  /**
+   * User request password reset
+   * Route: POST: /api/v1/user/request-password-reset
+   * @param {object} req - Request Object
+   * @param {object} res - Response Objec
+   * @returns {res} res - Response Object
+   * @memberof UserController
+   */
+  static async userRequestPasswordReset(req, res) {
+    let resetLinkURL;
+    if (process.env.NODE_ENV !== 'production') {
+      resetLinkURL = process.env.LOCAL_BASE_URL;
+    } else {
+      resetLinkURL = process.env.BASE_URL;
+    }
+
+    const { email } = req.body;
+    try {
+      const { foundUser } = req;
+      const payload = {
+        id: foundUser.id,
+        username: foundUser.username,
+        role: 1
+      };
+      const token = await Authentication.getToken(payload, 900000);
+      if (foundUser) {
+        const tokenCreated = await userTokens.create({
+          token,
+          userId: foundUser.id
+        });
+        resetLink = `${resetLinkURL}${tokenCreated.token}`;
+        const details = {
+          emailBody: `
+            Hi ${foundUser.firstName}, click ${resetLink} to reset password
+          `,
+          subject: 'Request to reset password',
+          email
+        };
+        await SendEmail.emailSender(details);
+        return res.status(200).send({
+          status: 'success',
+          message: 'check your mail for instructions on how to reset password'
+        });
+      }
+      return res.status(404).send({
+        status: 'error',
+        error: 'User with email not found'
+      });
+    } catch (error) {
+      return res.status(500).send({
+        message: 'Internal Server Error',
+        error
+      });
+    }
```

Comment:
@andrewinsoul we don't want to return server errors from the back-end to the front-end. This is a bad security practice because you are exposing the server to blind SQL injection attacks. I suggest removing the line 106 and just send the "Internal Server Error" message to the front-end.